airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/AsyncStreamConsumer.java

```diff
@@ -107,7 +107,9 @@ public void accept(final String messageString, final Integer sizeInBytes) throws
             validateRecord(message);
           }
 
-          bufferEnqueue.addRecord(message, sizeInBytes);
+          // The offset is to compensate for the overhead of PartialAirbyteMessage references
+          // which is 8 bytes per reference (64 bit JVM) and the size of the object size in bytes
+          bufferEnqueue.addRecord(message, sizeInBytes + 104);
```

Comment:
how did we end up with 104?

Reply:
104 is based on 8 bytes per reference and the number of referenced fields in PartialAirbyteMessage, PartialAirbyteStreamState, PartialRecordMessage, PartialStateMessage, and StreamDescriptor which totaled 13 references

Reply:
Yes, on it!

Reply:
Ran with 80 bytes and saw not only successful syncs but ever so slightly better sync times about 20 - 40 seconds for 12/13 min avg syncs. Great catch on the science!